src/lib.rs

```diff
@@ -1,123 +1,289 @@
+pub mod trust_store;
 pub mod types;
 pub mod utils;
-pub mod constants;
 
-use x509_parser::certificate::X509Certificate;
+use std::time::SystemTime;
 
-#[cfg(test)]
-mod tests {
-    use crate::types::tcbinfo::{TcbInfoV2, TcbInfoV3};
-    use crate::types::quotes::{version_4::QuoteV4, version_3::QuoteV3};
-    use crate::types::collaterals::IntelCollateral;
-
-    use crate::utils::cert::{hash_crl_keccak256, hash_x509_keccak256, parse_crl_der, parse_pem, parse_x509_der, verify_crl};
-    use crate::utils::tcbinfo::{validate_tcbinfov2, validate_tcbinfov3};
-    use crate::utils::quotes::{
-        version_3::verify_quote_dcapv3, 
-        version_4::verify_quote_dcapv4
+use anyhow::{anyhow, bail, Context};
+use p256::ecdsa::{signature::Verifier, VerifyingKey};
+use trust_store::TrustStore;
+use types::{
+    collateral::Collateral,
+    quote::{Quote, TDX_TEE_TYPE},
+    sgx_x509::SgxPckExtension,
+    tcb_info::{TcbInfo, TcbStanding},
+    VerifiedOutput,
+};
+use utils::Expireable;
+use zerocopy::AsBytes;
+
+pub fn verify_dcap_quote(
+    current_time: SystemTime,
+    collateral: Collateral,
+    quote: Quote,
+) -> anyhow::Result<VerifiedOutput> {
+    // 1. Verify the integrity of the signature chain from the Quote to the Intel-issued PCK
+    //    certificate, and that no keys in the chain have been revoked.
+    let tcb_info = verify_integrity(current_time, &collateral, &quote)?;
+
+    // 2. Verify the Quoting Enclave source and all signatures in the Quote.
+    verify_quote(&collateral, &quote)?;
+
+    // 3. Verify the status of Intel SGX TCB described in the chain.
+    let tcb_standing = verify_tcb_status(&tcb_info, &quote.signature.pck_extension)?;
+    let (advisory_ids, mut tcb_status) = match tcb_standing {
+        TcbStanding::UpToDate => (None, types::tcb_info::TcbStatus::UpToDate),
+        TcbStanding::SWHardeningNeeded { advisory_ids } => (
+            Some(advisory_ids),
+            types::tcb_info::TcbStatus::SWHardeningNeeded,
+        ),
     };
 
-    // Pinned September 10th, 2024, 6:49am GMT
-    // there's no need for constant sample collateral updates
-    const PINNED_TIME: u64 = 1725950994;
+    // 4. If TDX type then verify the status of TDX Module status and converge and send
+    if quote.header.tee_type == TDX_TEE_TYPE {
+        let tdx_module_status =
+            tcb_info.verify_tdx_module(quote.body.as_tdx_report_body().unwrap())?;
+        tcb_status = TcbInfo::convere_tcb_status_with_tdx_module(tcb_status, tdx_module_status);
+    }
 
```

Comment:
Final TCB convergence with the QE Identity TCB level is missing.

Reply:
Done https://github.com/automata-network/dcap-rs/pull/15/commits/d5d18c1d22783d36c99338af94d6446fe739aede